batchlib/analysis/pixel_level_analysis.py

```diff
@@ -0,0 +1,192 @@
+import matplotlib
+matplotlib.use("Agg")
+import matplotlib.pyplot as plt
+# import seaborn as sns
+from matplotlib.pyplot import imshow
+
+import pathlib
+import argparse
+import os
+import os.path
+from concurrent import futures
+from glob import glob
+import json
+import h5py
+import numpy as np
+from tqdm import tqdm
+from time import sleep
+from ..base import BatchJob, BatchJobWithSubfolder
+from ..util.plate_visualizations import well_plot
+
+
+def open_hdf5(filename, *args, **kwargs):
+    while True:
+        try:
+            hdf5_file = h5py.File(filename, *args, **kwargs)
+            break  # Success!
+        except OSError:
+            print(f"waiting on {filename}")
+            sleep(5)  # Wait a bit
+    return hdf5_file
+
+
+def load_sample(path):
+    with open_hdf5(path, mode='r') as f:
+        tritc = f['raw'][2]
+        local_infection_probs = f['local_infection'][()]
+
+    infected = local_infection_probs[0] > 0.5
+    not_infected = local_infection_probs[1] > 0.5
+
+    return infected, not_infected, tritc
+
+
+def create_folder(*split_path):
```

Comment:
maybe I don't quite understand what this does, but isn't this the same as `os.makedirs(..., exist_ok=True)`?

Reply:
Yes, it is the same as a *combination* of `os.path.join` and `os.makedirs`. 
Would you prefer `os.makedirs` over  `pathlib.Path(...).mkdir`?

Anyway, I just realized that I use this function exactly once in the final script. Therefore this seems very unnecessary. I will refactor it.